lib/bricks-json.js

```diff
@@ -1,21 +1,93 @@
 'use strict';
 
-var when = require('when');
+var inspect = require('util').inspect;
 
+var cppArguments = require('./cpp-arguments');
 
-exports.JSON = function (object, name) {
-	return when(object).then(function (ret) {
-		var object = ret;
-		if (typeof name === 'string') {
-			object = {};
-			object[name] = ret;
+var JavaScriptJSON = global.JSON;
+
+
+function Serializable() {
+	Object.defineProperties(this, {
+		serialize: {
+			configurable: false,
+			enumerable: false,
+			writable: false,
+			value: this.serialize || function (ar) {
+				var _this = this;
+				
+				Object.keys(_this).forEach(function (key) {
+					if (typeof _this[key] === 'function') {
+						// Ignore functions.
+						return;
+					}
+					
+					if (typeof _this[key] === 'object') {
+						if (Array.isArray(_this[key])) {
+							ar[key] = [].slice.call(_this[key]);
+						}
+						else if (_this[key] && typeof _this[key].serialize === 'function') {
+							ar[key] = {};
+							_this[key].serialize(ar[key]);
+						}
+						else {
+							// Ignore objects if they are not serializable.
+							return;
+						}
+					}
+					
+					ar[key] = _this[key];
+				});
+			}
 		}
-		return JSON.stringify(object);
 	});
-};
+}
 
-exports.ParseJSON = function (arg) {
-	return when(arg).then(function (ret) {
-		return JSON.parse(ret);
-	});
-};
+
+function JSON() {
+	return cppArguments.assert('JSON', [
+		[
+			cppArguments.assertion('object', 'T&&', 'object'),
+			cppArguments.assertion('string', 'const std::string&', 'name', cppArguments.ASSERTION_MODE_OPTIONAL),
+			function (object, name) {
+				if (typeof object.serialize !== 'function') {
+					throw new Error('JSON: Object is not serializable: ' + inspect(object));
+				}
+				
+				var ar = {};
+				
+				if (typeof name === 'string') {
```

Comment:
Why not `var field_name = (typeof name === 'string') ? name : 'value0');` and the rest in two lines?